src/lib/swapper/swappers/utils/filterBuyAssetsBySellAssetId/filterBuyAssetsBySellAssetId.ts

```diff
@@ -1,27 +1,58 @@
-import type { AssetId } from '@shapeshiftoss/caip'
+import type { Asset } from '@shapeshiftoss/asset-service'
+import type { AssetId, ChainId } from '@shapeshiftoss/caip'
 import { isEvmChainId } from '@shapeshiftoss/chain-adapters'
 import type { BuyAssetBySellIdInput } from 'lib/swapper/api'
 import { selectAssets } from 'state/slices/selectors'
 import { store } from 'state/store'
 
-// we dont perform a lookup to lifi's supported assets because they support far more assets than we do
-// so the overhead in performing the fetch to lifi isnt worth the time
-export function filterSameChainEvmBuyAssetsBySellAssetId(input: BuyAssetBySellIdInput): AssetId[] {
+const getSameChainEvmAssetsPredicate =
+  (sellAssetChainId: ChainId, assets: Partial<Record<AssetId, Asset>>) =>
+  (assetId: AssetId): boolean => {
+    const buyAsset = assets[assetId]
+
+    if (buyAsset === undefined) return false
+
+    // same-chain swaps and evm only
+    return buyAsset.chainId === sellAssetChainId && isEvmChainId(buyAsset.chainId)
+  }
+
+const getCrossChainEvmAssetsPredicate =
+  (sellAssetChainId: ChainId, assets: Partial<Record<AssetId, Asset>>) =>
+  (assetId: AssetId): boolean => {
+    const buyAsset = assets[assetId]
+
+    if (buyAsset === undefined) return false
+
+    // cross-chain swaps and evm only
+    return buyAsset.chainId !== sellAssetChainId && isEvmChainId(buyAsset.chainId)
+  }
```

Comment:
This could compose by re-using most of the logic in both except for the `buyAsset.chainId !== sellAssetChainId` / `buyAsset.chainId === sellAssetChainId` variant if you feel like it, obviously fine without

Reply:
yeah agreed, i've refactored a bit and now its a fair bit simpler